packages/bff/src/services/agreementService.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable functional/immutable-data */
+import {
+  AgreementProcessApiAgreement,
+  AgreementProcessApiResponse,
+} from "../model/api/agreementTypes.js";
+import { BffGetCatalogApiHeaders } from "../model/api/bffTypes.js";
+import { EServiceCatalogProcessApi } from "../model/api/catalogTypes.js";
+import { AgreementProcessClient } from "../providers/clientProvider.js";
+
+export const getLatestAgreement = async (
+  agreementProcessClient: AgreementProcessClient,
+  consumerId: string,
+  eservice: EServiceCatalogProcessApi,
+  headers: BffGetCatalogApiHeaders
+): Promise<AgreementProcessApiAgreement> => {
+  const getAgreementsFrom = async (
+    start: number
+  ): Promise<AgreementProcessApiResponse> =>
+    await agreementProcessClient.getAgreements({
+      headers,
+      queries: {
+        consumersIds: consumerId,
+        eservicesIds: eservice.id,
+        offset: start,
+        limit: 50,
+      },
+    });
+
+  // Fetched all agreements in a recursive way
+  const getAgreements = async (
+    start: number
+  ): Promise<AgreementProcessApiAgreement[]> => {
+    const agreements = (await getAgreementsFrom(start)).results;
+
+    if (agreements.length >= 50) {
+      return agreements.concat(await getAgreements(start + 50));
+    }
+    return agreements;
+  };
+
+  const allAgreements = await getAgreements(0);
+
+  return allAgreements.sort((firstAgreement, secondAgreement) => {
+    if (firstAgreement.version !== secondAgreement.version) {
```

Comment:
here shouldn't be the opposite? We should check the date only if the `version` is the same

Reply:
It should be correct, in this case the lambda function return a boolean results of criteria adopt for ordering the current values, in this case we want apply greater version number criteria, if the versions between agreement are different. If agreement version number are equals we should adopt greater version criteria on agreement's descriptors.

Reply:
I'M WRONG! 
you are right this condition should order by version and not agreement `createDate` , fixed it with https://github.com/pagopa/interop-be-monorepo/pull/665/commits/b7874c52a39afcd8bebc9779eff527641dbab4c8